docs/examples/use_cases/tensorflow/efficientdet/pipeline/dali/ops.py

```diff
@@ -0,0 +1,216 @@
+
+
+    images = dali.fn.decoders.image(
+        encoded,
+        device="mixed" if device == "gpu" else "cpu",
+        output_type=dali.types.RGB,
+    )
+
+    shape = dali.fn.peek_image_shape(encoded)
+    h = dali.fn.slice(shape, 0, 1, axes=(0,))
+    w = dali.fn.slice(shape, 1, 1, axes=(0,))
+
+    return (
+        images,
+        bboxes,
+        classes,
+        dali.fn.cast(w, dtype=dali.types.FLOAT),
+        dali.fn.cast(h, dtype=dali.types.FLOAT),
+    )
+
+
+def normalize_flip(device, images, bboxes, p=0.5):
+    flip = dali.fn.random.coin_flip(probability=p, device="cpu")
+    images = dali.fn.crop_mirror_normalize(
+        images,
+        mirror=flip.gpu() if device == "gpu" else flip,
+        mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+        std=[0.229 * 255, 0.224 * 255, 0.225 * 255],
+        output_layout=dali.types.NHWC,
+        device=device,
+    )
+    bboxes = dali.fn.bb_flip(bboxes, horizontal=flip, ltrb=True, device="cpu")
+    return images, bboxes
+
+
+def gridmask(device, images, widths, heights):
+    if device == "gpu":
+        widths = widths.gpu()
+        heights = heights.gpu()
+
+    ratio = 0.4
+    angle = (
+        dali.fn.random.normal(mean=-1, stddev=1, device=device)
+        * 10.0
+        * (math.pi / 180.0)
+    )
+
+    l = dali.math.min(0.5 * heights, 0.3 * widths, device=device)
+    r = dali.math.max(0.5 * heights, 0.3 * widths, device=device)
+    tile = dali.fn.cast(
+        (dali.fn.random.uniform(range=[0.0, 1.0], device=device) * (r - l) + l),
+        dtype=dali.types.INT32,
+        device=device,
+    )
+
+    gridmask = dali.fn.grid_mask(
+        images, ratio=ratio, angle=angle, tile=tile, device=device
+    )
+
+    p = dali.fn.random.coin_flip(device=device)
+    images = images * (1 - p) + gridmask * p
+
+    return images
+
+
+def random_crop_resize(
+    device, images, bboxes, classes, widths, heights, output_size, scaling=[0.1, 2.0]
+):
+
+    if scaling is None:
+        scale_factor = 1.0
+    else:
+        scale_factor = dali.fn.random.uniform(range=scaling, device="cpu")
+
+    scaled_x = scale_factor * output_size[0]
+    scaled_y = scale_factor * output_size[1]
+
+    image_scale = dali.math.min(scaled_x / widths, scaled_y / heights)
+
+    scaled_widths = widths * image_scale
+    scaled_heights = heights * image_scale
+
+    images = dali.fn.resize(
+        images,
+        resize_x=scaled_widths.gpu() if device == "gpu" else scaled_widths,
+        resize_y=scaled_heights.gpu() if device == "gpu" else scaled_heights,
+        device=device,
+    )
+
+    crop_shape = dali.fn.constant(idata=output_size, device="cpu")
+
+    anchors, shapes, bboxes, classes = dali.fn.random_bbox_crop(
+        bboxes,
+        classes,
+        crop_shape=crop_shape,
+        input_shape=dali.fn.cast(
+            dali.fn.cat(scaled_widths, scaled_heights, device="cpu"),
+            dtype=dali.types.INT32,
+            device="cpu",
+        ),
```

Comment:
```suggestion
        input_shape=dali.fn.cast(dali.fn.cat(scaled_widths, scaled_heights), dtype=dali.types.INT32),```
...or better still - have `sizes`, as noted above.